The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11930764398/job/33252212009) [(plain)](https://github.com/rust-lang/rust/commit/a8fded6051dd9eee167e7eb7331209e139bf1e10/checks/33252212009/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.720 Building wheels for collected packages: reuse
#13 2.721   Building wheel for reuse (pyproject.toml): started
#13 2.968   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.968   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 2.969   Stored in directory: /tmp/pip-ephem-wheel-cache-iw9358ar/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.971 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.363 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.363 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.891 Collecting virtualenv
#13 3.891 Collecting virtualenv
#13 3.927   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 3.984      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 57.4 MB/s eta 0:00:00
#13 4.021 Collecting distlib<1,>=0.3.7
#13 4.024   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.032      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 77.2 MB/s eta 0:00:00
#13 4.065 Collecting filelock<4,>=3.12.2
#13 4.068   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.101 Collecting platformdirs<5,>=3.9.1
#13 4.104   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.187 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.365 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      325568 kB
DirectMap2M:     9111552 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 30.43s
##[endgroup]
fmt check
Diff in /checkout/library/std/src/lib.rs:658:
     pub use std_detect::is_aarch64_feature_detected;
     #[unstable(feature = "stdarch_arm_feature_detection", issue = "111190")]
     pub use std_detect::is_arm_feature_detected;
-    #[unstable(feature = "is_riscv_feature_detected", issue = "111192")]
-    pub use std_detect::is_riscv_feature_detected;
     #[unstable(feature = "is_loongarch_feature_detected", issue = "117425")]
     pub use std_detect::is_loongarch_feature_detected;
+    #[unstable(feature = "is_riscv_feature_detected", issue = "111192")]
+    pub use std_detect::is_riscv_feature_detected;
     #[stable(feature = "simd_x86", since = "1.27.0")]
     pub use std_detect::is_x86_feature_detected;
     #[unstable(feature = "stdarch_mips_feature_detection", issue = "111188")]
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/openbsd/raw.rs" "/checkout/library/std/src/os/openbsd/mod.rs" "/checkout/library/std/src/os/l4re/fs.rs" "/checkout/library/std/src/os/l4re/raw.rs" "/checkout/library/std/src/os/l4re/mod.rs" "/checkout/library/std/src/os/netbsd/fs.rs" "/checkout/library/std/src/os/netbsd/net.rs" "/checkout/library/std/src/os/netbsd/raw.rs" "/checkout/library/std/src/os/netbsd/mod.rs" "/checkout/library/std/src/os/illumos/fs.rs" "/checkout/library/std/src/os/illumos/raw.rs" "/checkout/library/std/src/os/illumos/mod.rs" "/checkout/library/std/src/os/nuttx/fs.rs" "/checkout/library/std/src/os/nuttx/raw.rs" "/checkout/library/std/src/os/nuttx/mod.rs" "/checkout/library/std/src/os/rtems/fs.rs" "/checkout/library/std/src/os/rtems/raw.rs" "/checkout/library/std/src/os/rtems/mod.rs" "/checkout/library/std/src/os/uefi/env.rs" "/checkout/library/std/src/os/uefi/mod.rs" "/checkout/library/std/src/os/aix/fs.rs" "/checkout/library/std/src/os/aix/raw.rs" "/checkout/library/std/src/os/aix/mod.rs" "/checkout/library/std/src/os/linux/fs.rs" "/checkout/library/std/src/os/linux/process.rs" "/checkout/library/std/src/os/linux/net.rs" "/checkout/library/std/src/os/linux/raw.rs" "/checkout/library/std/src/os/linux/mod.rs" "/checkout/library/std/src/os/horizon/fs.rs" "/checkout/library/std/src/os/horizon/raw.rs" "/checkout/library/std/src/os/horizon/mod.rs" "/checkout/library/std/src/os/mod.rs" "/checkout/library/std/src/random.rs" "/checkout/library/std/src/path.rs" "/checkout/library/std/src/backtrace/tests.rs" "/checkout/library/std/src/lib.miri.rs" "/checkout/library/std/src/f16/tests.rs" "/checkout/library/std/src/prelude/common.rs" "/checkout/library/std/src/prelude/mod.rs" "/checkout/library/std/src/process/tests.rs" "/checkout/library/std/src/time.rs" "/checkout/library/std/src/time/tests.rs" "/checkout/library/std/src/f32/tests.rs" "/checkout/library/std/src/alloc.rs" "/checkout/library/std/src/lib.rs" "/checkout/library/profiler_builtins/build.rs" "/checkout/library/profiler_builtins/src/lib.rs" "/checkout/library/rustc-std-workspace-core/lib.rs" "/checkout/library/rustc-std-workspace-std/lib.rs" "/checkout/library/unwind/src/libunwind.rs" "/checkout/library/unwind/src/unwinding.rs" "/checkout/library/unwind/src/wasm.rs" "/checkout/library/unwind/src/lib.rs" "/checkout/tests/run-make/emit-to-stdout/rmake.rs" "/checkout/tests/run-make/rust-lld-by-default-nightly/main.rs" "/checkout/tests/run-make/rust-lld-by-default-nightly/rmake.rs" "/checkout/tests/run-make/optimization-remarks-dir/foo.rs" "/checkout/tests/run-make/optimization-remarks-dir/rmake.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/rmake.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/src/lib.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/examples/ex.rs" "/checkout/tests/run-make/rustc-help/rmake.rs" "/checkout/tests/run-make/ice-static-mir/foo.rs" "/checkout/library/std/src/os/openbsd/fs.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Nov 20 09:55:52 UTC 2024
  network time: Wed, 20 Nov 2024 09:55:52 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -658,6 +658,8 @@ pub mod arch {
     pub use std_detect::is_aarch64_feature_detected;
     #[unstable(feature = "stdarch_arm_feature_detection", issue = "111190")]
     pub use std_detect::is_arm_feature_detected;
+    #[unstable(feature = "is_loongarch_feature_detected", issue = "117425")]
+    pub use std_detect::is_loongarch_feature_detected;
     #[unstable(feature = "is_riscv_feature_detected", issue = "111192")]
     pub use std_detect::is_riscv_feature_detected;
     #[stable(feature = "simd_x86", since = "1.27.0")]

